Helpers/TaskHelpers.cs

```diff
@@ -9,6 +9,75 @@ namespace CRCTray.Helpers
 {
    internal class TaskHelpers
    {
+       public static async Task<T> TryTask<T>(Func<T> function)
+       {
+            try
+            {
+                return await Task.Run(function);
+            }
+            catch (AggregateException ae)
+            {
+                ae.Handle((x) =>
+                {
+                    if (x is APICommunicationException) // This we know how to handle.
+                    {
+                        // TODO: start counting and eventually notify
+
+                        return true;
+                    }
+
+                    if (x is APIException) // This we know how to handle.
+                    {
+                        return true;
+                    }
+
+                    if (x is TaskCanceledException)

```

Comment:
`TimeoutException` will not occur here?

Reply:
It was probably missed, as it is only in `public static async Task<T> TryTaskAndNotify<T, TArgs>` where it includes `Targs`.

Follow-up refactor as it needs to be added in two places.